### PR TITLE
Complete send data and receive data from api functionality

### PR DESCRIPTION
**Pull request by jubaer919**
# Highlight 🔍 : 
- Create a new game with the name of your choice by using the API
  - Make sure that you saved the ID of your game that will be returned by API.
- Implement the "Refresh" button that will get all scores for a game created by you from the API
- Use `arrow functions` instead of the `function` keyword.
- Use `async` and `await` JavaScript features to consume the API.

### CODE REVIEW

File: src/leaderboard.js [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    [, , , gameID] = response.data.result.split(' ');
+
+    return gameID;
+  } catch (error) {
+    const h2 = document.createElement('h2');
+    h2.innerText = '404 error hapend';
+    body.appendChild(h2);
+    throw error;
+  }
+}
```
Comment (Whoistolu) [inline src/leaderboard.js]: - [ ] [REQUIRED] Let's remember to ensure the usage of **arrow functions** as stated by the project requirement. This is a regular async function declaration. Arrow functions have a different syntax, using the => arrow notation.

Here is an equivalent version of this code using an arrow function :
``` javascript
const submitScore = async (user, score) => {
  try {
    const data = {
      user,
      score,
    };
    const url = `https://us-central1-js-capstone-backend.cloudfunctions.net/api/games/${gameID}/scores/`;
    await axios.post(url, data);

    // Clear input fields
    document.querySelector('.name-input').value = '';
    document.querySelector('.number-input').value = '';
  } catch (error) {
    const h2 = document.createElement('h2');
    h2.innerText = '404 error happened';
    body.appendChild(h2);
  }
};

Kindly fix this 👍🏽 
```

File: src/leaderboard.js [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+      listItem.classList.add('list-items');
+      listItem.innerHTML = `<span class="span-name">${person.user}:</span>  <span class="span-number">${person.score}</span>`;
+      list.appendChild(listItem);
+    });
+  } catch (error) {
+    const h2 = document.createElement('h2');
+    h2.innerText = '404 error hapend';
+    body.appendChild(h2);
+  }
+}
```
Comment (Whoistolu) [inline src/leaderboard.js]: - [ ] [REQUIRED] Let's remember to ensure the usage of **arrow functions** as stated by the project requirement. This is a regular async function declaration. Arrow functions have a different syntax, using the => arrow notation.
Kindly fix this 👍🏽

File: src/leaderboard.js [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+    // Clear input fields
+    document.querySelector('.name-input').value = '';
+    document.querySelector('.number-input').value = '';
+  } catch (error) {
+    const h2 = document.createElement('h2');
+    h2.innerText = '404 error hapend';
+    body.appendChild(h2);
+  }
+}
```
Comment (Whoistolu) [inline src/leaderboard.js]: - [ ] [REQUIRED] Let's remember to ensure the usage of **arrow functions** as stated by the project requirement. This is a regular async function declaration. Arrow functions have a different syntax, using the => arrow notation.
Kindly fix this 👍🏽

File: src/leaderboard.js [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+      }
+
+      submitScore(user, score);
+    });
+  } catch (error) {
+    const h2 = document.createElement('h2');
+    h2.innerText = '404 error hapend';
+    body.appendChild(h2);
+  }
+}
```
Comment (Whoistolu) [inline src/leaderboard.js]: - [ ] [REQUIRED] Let's remember to ensure the usage of **arrow functions** as stated by the project requirement. This is a regular async function declaration. Arrow functions have a different syntax, using the => arrow notation.
Kindly fix this 👍🏽

File: src/leaderboard.js [hunk 5]
```diff
@@ -0,0 +1,90 @@
+import axios from 'axios';
+
+let gameID;
+const body = document.querySelector('body');
+
+async function createGame(name) {
+  const baseURL = 'https://us-central1-js-capstone-backend.cloudfunctions.net/api/games/';
+
+  try {
+    const response = await axios.post(baseURL, { name });
```
Comment (Whoistolu) [inline src/leaderboard.js:10]: -  [ ] [REQUIRED] This code does not use async and await keywords with the fetch method. Instead, it uses the axios library for making HTTP requests. Kindly ensure you stick to the requirements of the project by using async and await keywords with the fetch method.

**Review by Whoistolu — CHANGES_REQUESTED** [review]
Hi @jubaer919 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️

- [ ]  _Check the inline comments under the review for sections to improve._

### Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

### N/A

Cheers and Happy coding!

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Shedrack-Sunday — APPROVED** [review]
Hi @jubaer919 👋

Congratulations! 🎉 Your project is complete and ready to be merged! 🚀

image


## Highlights
- Good in job with the changes in use of arrow functions ✔️ 
- Kudus to meet the exact project requirements   👍 

## Optional Suggestions 💡
While these comments with the [OPTIONAL] prefix won't prevent the approval of this pull request, I strongly recommend taking them into account as they can improve your code. Some of them were missed by the previous reviewer and addressing them will really enhance your application.
Cheers, and Happy coding!👏👏👏

If you have any questions or need clarification on anything, feel free to leave a comment in the pull request thread, and don't forget to tag me so I'll receive the notification.

As per the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561), you have a limited number of reviews per project. If you feel that the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request closed without merging**